### PR TITLE
Loader - get image dictionary

### PR DESCRIPTION
**Pull request by codyly**
Add `get_image_dictionary` to `loader_base` to get image dictionary directly from loader.

### CODE REVIEW

File: gtsfm/loader/loader_base.py [hunk 1]
```diff
@@ -143,3 +143,12 @@ def get_valid_pairs(self) -> List[Tuple[int, int]]:
                     indices.append((idx1, idx2))
 
         return indices
+
+    def get_image_dictionary(self) -> Dict[int, Image]:
```
Comment (johnwlambert) [inline gtsfm/loader/loader_base.py:147]: looks good @codyly . Can you add a unit test on the Lund door set 1 data in tests/data?

File: gtsfm/loader/loader_base.py [hunk 2]
```diff
@@ -143,3 +143,12 @@ def get_valid_pairs(self) -> List[Tuple[int, int]]:
...
+        """Get the dictionary containing all loaded images. The keys are the images' indices.
+
+        Returns:
+            Dict[int, Image]: image dictionary
```
Comment (johnwlambert) [inline gtsfm/loader/loader_base.py:151]: No need to repeat the type her, so I would prefer
```python

Returns:
    dictionary mapping image indices [0,...,N-1] to their associated image
```

File: gtsfm/loader/loader_base.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/loader/loader_base.py:147]: actually, we already have this functionality in 
```
            image_graph=loader.create_computation_graph_for_images(),
```
right?

Comment (johnwlambert) [reply gtsfm/loader/loader_base.py:147]: so I dont think we need this PR, after all, if we are working within Dask

Comment (codyly) [reply gtsfm/loader/loader_base.py:147]: Ah...I think so. I am not very familiar with Dask so I didn't realize it.

Comment (johnwlambert) [reply gtsfm/loader/loader_base.py:147]: where are you hoping to use this function @codyly ?

Comment (codyly) [reply gtsfm/loader/loader_base.py:147]: in the densify implementation at #195

Comment (codyly) [reply gtsfm/loader/loader_base.py:147]: After we get `sfm_result` in `run_scene_optimizer`, `sfm_result` together with `images_dict` will be passed to `MVSPatchmatchnet.densify` (implemented in #195)

Comment (codyly) [reply gtsfm/loader/loader_base.py:147]: @johnwlambert As the `MVSBase` defines the densify API as
```def densify(self, images: Dict[int, Image], sfm_result: GtsfmData) -> np.ndarray:```
Is there a simple way to transfter image list from `loader.create_computation_graph_for_images()` to image dictionary, such as
```dict(zip(list(range(len(loader))), loader.create_computation_graph_for_images()))```

**Comment by ayushbaid** [comment]
@codyly do we need this PR?

**Comment by codyly** [comment]
@ayushbaid @johnwlambert perhaps we need to find a proper approach to invoke PatchmatchNet in gtsfm runners. Currently it can be
```python
from gtsfm.densify.mvs_patchmatchnet import MVSPatchmatchNet

img_dict = {i: self.loader.get_image(i) for i in range(len(self.loader))}
points = MVSPatchmatchNet().densify(img_dict, sfm_result)
```

**Comment by ayushbaid** [comment]
> @ayushbaid @johnwlambert perhaps we need to find a proper approach to invoke PatchmatchNet in gtsfm runners. Currently it can be
> ```python
> from gtsfm.densify.mvs_patchmatchnet import MVSPatchmatchNet
> 
> img_dict = {i: self.loader.get_image(i) for i in range(len(self.loader))}
> points = MVSPatchmatchNet().densify(img_dict, sfm_result)
> ```

Yep this should be good!

**Pull request closed without merging**